### PR TITLE
r2mod_cli: init at 1.0.5

### PR DESCRIPTION
**Pull request by reedrw**
###### Motivation for this change
add [r2mod_cli](https://github.com/foldex/r2mod_cli), a Risk of Rain 2 mod manager written in Bash.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/r2mod_cli/default.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+  phases = [ "installPhase" ];
+
+  installPhase = ''
+    mkdir -p "$out/bin"
+    mkdir -p "$out/share/bash-completion/completions"
+    mkdir -p "$out/share/zsh/site-functions"
+    install -m 755 "$src/r2mod" "$out/bin/r2mod"
+    install -m 644 "$src/completions/bash/r2mod.sh" "$out/share/bash-completion/completions/r2mod"
+    install -m 644 "$src/completions/zsh/_r2mod" "$out/share/zsh/site-functions/_r2mod"
```
Comment (rissson) [inline pkgs/games/r2mod_cli/default.nix:34]: You can combine all of those with `install -D`, which creates the missing directories

Comment (reedrw) [reply pkgs/games/r2mod_cli/default.nix]: Thanks for pointing this out. :+1:

File: pkgs/games/r2mod_cli/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    repo = "r2mod_cli";
+    rev = "v${version}";
+    sha256 = "1g64f8ms7yz4rzm6xb93agc08kh9sbwkhvq35dpfhvi6v59j3n5m";
+
+  };
+
+  buildInputs = [
+    jq
+    makeWrapper
+    p7zip
```
Comment (rissson) [inline pkgs/games/r2mod_cli/default.nix:23]: You shouldn't need `p7zip` neither `jq` here.

**Review by rissson — CHANGES_REQUESTED** [review]

**Comment by rissson** [comment]
Result of `nixpkgs-review pr 105391` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>r2mod_cli</li>
  </ul>
</details>

File: pkgs/games/r2mod_cli/default.nix (hunk 2 above)
Comment (reedrw) [reply pkgs/games/r2mod_cli/default.nix]: Sure enough.

**Review by rissson — APPROVED** [review]

File: pkgs/games/r2mod_cli/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  pname = "r2mod_cli";
+  version = "1.0.5";
+
+  src = fetchFromGitHub {
+    owner = "Foldex";
+    repo = "r2mod_cli";
+    rev = "v${version}";
+    sha256 = "1g64f8ms7yz4rzm6xb93agc08kh9sbwkhvq35dpfhvi6v59j3n5m";
+
+  };
```
Comment (SuperSandro2000) [inline pkgs/games/r2mod_cli/default.nix:18]:
```suggestion
    sha256 = "1g64f8ms7yz4rzm6xb93agc08kh9sbwkhvq35dpfhvi6v59j3n5m";
  };
```

File: pkgs/games/r2mod_cli/default.nix [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+  buildInputs = [ makeWrapper ];
+
+  phases = [ "installPhase" ];
```
Comment (SuperSandro2000) [inline pkgs/games/r2mod_cli/default.nix:22]:
```suggestion
```

Please do not overwrite phases.

File: pkgs/games/r2mod_cli/default.nix [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+  meta = with stdenv.lib; {
+    description = "A Risk of Rain 2 Mod Manager in Bash";
+    homepage = "https://github.com/foldex/r2mod_cli";
+    license = licenses.gpl3;
+    maintainers = [ maintainers.reedrw ];
+    platforms = platforms.linux;
+  };
+
+}
```
Comment (SuperSandro2000) [inline pkgs/games/r2mod_cli/default.nix:41]:
```suggestion
  };
}
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105391` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>r2mod_cli</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
```
building
build flags: SHELL=/nix/store/9yapmh2s4z7dz07zdpd626329bcqh1p0-bash-4.4-p23/bin/bash
install -d "/usr/bin"
install: cannot create directory '/usr': Permission denied
make: *** [Makefile:5: install] Error 1
```

**Comment by reedrw** [comment]
This is happening because it's running the [makefile](https://github.com/Foldex/r2mod_cli/blob/master/Makefile) from the source directory (which only contains an install step). What's wrong with just overwriting phases?

**Comment by reedrw** [comment]
I assume the more idiomatic way to do it is to use make flags?

**Comment by SuperSandro2000** [comment]
> What's wrong with just overwriting phases?

Some hooks expect phases to contain certain code and overwriting them breaks the hooks.

File: pkgs/games/r2mod_cli/default.nix [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  };
+
+  buildInputs = [ makeWrapper ];
+
+  makeFlags = [ "PREFIX=$(out)" ];
+
+  postInstall = ''
+    wrapProgram $out/bin/r2mod \
+      --prefix PATH : ${jq}/bin \
+      --prefix PATH : ${p7zip}/bin
```
Comment (SuperSandro2000) [inline pkgs/games/r2mod_cli/default.nix:26]:
```suggestion
    wrapProgram $out/bin/r2mod --prefix PATH : "${stdenv.lib.makeBinPath [ jq p7zip ]}";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/games/r2mod_cli/default.nix [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  makeFlags = [ "PREFIX=$(out)" ];
+
+  postInstall = ''
+    wrapProgram $out/bin/r2mod --prefix PATH : "${stdenv.lib.makeBinPath [ jq p7zip ]}";
+  '';
+
+  meta = with stdenv.lib; {
+    description = "A Risk of Rain 2 Mod Manager in Bash";
+    homepage = "https://github.com/foldex/r2mod_cli";
+    license = licenses.gpl3;
```
Comment (SuperSandro2000) [inline pkgs/games/r2mod_cli/default.nix:30]: gpl3 is ambiguous. Please use either gpl3Plus or gpl3Only.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105391` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>r2mod_cli</li>
  </ul>
</details>

**Pull request merged**